site/docs/packages/recipes.md

```diff
@@ -118,6 +118,59 @@ export const button = recipe({
 });
 ```
 
+The recipes function also exposes an array property `variants` that includes all the variants from your recipe.
+
+```ts
+button.variants();
+// -> ['color', 'size']
+```
+
+## Recipe class name selection
+
+Recipes function exposes internal class names in `classNames` property.
+The property has two predefined props: `base` and `variants`. The `base` prop includes base class name. It is always defined even if you do not have any base styles. The `variants` prop includes class names for each defined variant.
+
+```ts
+// app.css.ts
+console.log(button.classNames.base);
+// -> app_button__129pj250
+console.log(button.classNames.variants.color.neutral);
+// -> app_button_color_neutral__129pj251
+console.log(button.classNames.variants.size.small);
+// -> app_button_size_small__129pj254
+```
+
+> 🚧&nbsp;&nbsp;The class names do not have `.` at the beginning.
+
+Exposed class names allow you to select special variant in cascade styles:
+
+```ts compiled
+// button.css.ts
+import { style } from '@vanilla-extract/css';
+import { recipe } from '@vanilla-extract/recipes';
+
+export const button = recipe({
+  base: { fontWeight: 'bold' },
+  variants: {
+    color: {
+      neutral: { background: 'transparent' }
+    }
+  }
+});
+
+export const parent = style({
+  background: 'blue',
+  selectors: {
+    [`.${button.classNames.base} &`]: {
+      padding: 10
+    },
+    [`.${button.classNames.variants.color.neutral} &`]: {
+      color: 'red'
+    }
+  }
+});
+```
+
```

Comment:
```suggestion
> 🚧&nbsp;&nbsp;The class names do not have `.` at the beginning.

Exposed class names allow you to select special variant in cascade styles:

```ts compiled
// button.css.ts
import { style } from '@vanilla-extract/css';
import { recipe } from '@vanilla-extract/recipes';

export const button = recipe({
  base: { fontWeight: 'bold' },
  variants: {
    color: {
      neutral: { background: 'transparent' }
    }
  }
});

export const parent = style({
  background: 'blue',
  selectors: {
    [`.${button.classNames.base} &`]: {
      padding: 10
    },
    [`.${button.classNames.variants.color.neutral} &`]: {
      color: 'red'
    }
  }
});
```

```

Reply:
VE has a system for automatically handling this case so you don't need to add a `.` yourself.